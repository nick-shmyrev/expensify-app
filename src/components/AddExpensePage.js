import React from 'react';
import { connect } from 'react-redux';
import { addExpense } from '../actions/expenses-actions';
import ExpenseForm from './ExpenseForm';

export class AddExpensePage extends React.Component {
  onExpenseFormSubmit = (expense) => {
    this.props.addExpense(expense);
    this.props.history.push('/');
  };
  
  render() {
    return (
      <div>
        <h2>Add Expense</h2>
        <ExpenseForm
          onExpenseFormSubmit={this.onExpenseFormSubmit}
        />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  addExpense: expense => dispatch(addExpense(expense)),
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
