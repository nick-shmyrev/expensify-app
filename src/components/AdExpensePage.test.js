import React from 'react';
import { shallow } from 'enzyme';
import { AddExpensePage } from './AddExpensePage';

describe('AddExpensePage component', () => {
  let addExpense;
  let history;
  let wrapper;
  
  beforeEach(() => {
    addExpense = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(<AddExpensePage addExpense={addExpense} history={history}/>);
  });
  
  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  test('should handle onExpenseFormSubmit', () => {
    const expense = {
      description: 'Description',
      amount: 39.5,
      note: 'A note',
      createdAt: 0,
    };
    
    wrapper.find('ExpenseForm').prop('onExpenseFormSubmit')(expense);
    
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(addExpense).toHaveBeenLastCalledWith(expense);
  });
});
