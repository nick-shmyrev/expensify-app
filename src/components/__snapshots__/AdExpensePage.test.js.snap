// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddExpensePage component should render correctly 1`] = `
<div>
  <h2>
    Add Expense
  </h2>
  <ExpenseForm
    onExpenseFormSubmit={[Function]}
  />
</div>
`;
